components/button_selector.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+type Props = {
+    id: string;
+    value: string | number | boolean | null;
+    onChange?: (id: string, value: string) => void;
+    label?: React.ReactNode;
+    labelClassName?: string;
+    inputClassName?: string;
+    helpText?: React.ReactNode;
+    footer?: React.ReactNode;
+    disabled?: boolean;
+    shouldSubmit?: boolean;
+    options?: {text: string; value: string}[] | null;
+}
+const defaultProps: Partial<Props> = {
+    value: '',
+    labelClassName: '',
+    inputClassName: '',
+}
```

Comment:
Should `options` be marked as a required parameter? If not, maybe we should include a default value for it here like `options: []`